src/background/utils.ts

```diff
@@ -37,20 +37,22 @@ export async function getTabIdWithSearchOpen(
 export async function getTabsInCurrentWindow() {
   // should it return the current tab??
   const tabs = await browser.tabs.query({ currentWindow: true });
-  return tabs
-    .filter((tab) => {
-      // will all pages have a title?
-      return tab.id && tab.id !== browser.tabs.TAB_ID_NONE && tab.url;
-    })
-    .map(({ id, favIconUrl, title, url }) => {
+  const results: TabData[] = [];
+
+  for (let i = 0; i < tabs.length; i++) {
+    // will all pages have a title?
+    if (tabs[i].id && tabs[i].id !== browser.tabs.TAB_ID_NONE && tabs[i].url) {
```

Comment:
You could set `tabs[i]` to a variable so that it is easier to reference in other parts of the code.